src/build_queue.rs

```diff
@@ -49,15 +48,17 @@ impl BuildQueue {
         }
     }
 
-    pub fn last_seen_reference(&self) -> Result<Option<Oid>> {
+    pub fn last_seen_reference(&self) -> Result<Option<crates_index_diff::git::ObjectId>> {
         let mut conn = self.db.get()?;
         if let Some(value) = get_config::<String>(&mut conn, ConfigName::LastSeenIndexReference)? {
-            return Ok(Some(Oid::from_str(&value)?));
+            return Ok(Some(crates_index_diff::git::ObjectId::from_hex(
+                value.as_bytes(),
+            )?));
         }
         Ok(None)
     }
 
-    fn set_last_seen_reference(&self, oid: Oid) -> Result<()> {
+    fn set_last_seen_reference(&self, oid: &crates_index_diff::git::oid) -> Result<()> {
```

Comment:
nit, consistency: why is this `git::oid` while we return `git::ObjectId` above in the getter?

Reply:
The `git::oid` is akin to `std::path::Path` and `git::ObjectId` is akin to `std::path::PathBuf`. Thus it's considered 'better' to take references to object ids if they suffice. Note that `git::ObjectId` is also `Copy` as it's currently only 20 bytes, and I changed the parameter to `git::ObjectId` to resolve the issue in 3cc50c3fc3658fca3726e6149c2ecac4b101ab7a .